c2rust-analyze/src/rewrite/ty.rs

```diff
@@ -328,10 +433,19 @@ impl<'a, 'tcx> HirTyVisitor<'a, 'tcx> {
             self.hir_rewrites.push((hir_ty.span, rw));
         }
 
+        if let TyKind::Adt(adt_def, substs) = rw_lty.ty.kind() {
+            if !rw_lty.label.lifetime.is_empty() {
+                self.hir_rewrites.push((
```

Comment:
I was expecting this to generate a rewrite on every ADT with at least one lifetime, which would lead to unnecessary unfolding of type aliases.  But I didn't see that effect when I ran it on this code:

```Rust
struct Test<'a>(&'a u8);
type Alias<'a> = Test<'a>;
fn f<'a>(x: Alias<'a>) {}
```

I was expecting a rewrite from `x: Alias<'a>` to `x: Test<'a>`, but no such rewrite occurred.  Under what conditions is `label.lifetime` non-empty?  I was expecting it to be set to the original lifetime args `['a]` in this case.  Is it empty for structs that have no hypothetical lifetimes?

Reply:
`Alias` doesn't get rewritten because it doesn't appear in the ADT metadata table, and that's because it's `DefKind` is a `TyAlias` and not one of `Struct | Enum | Union`

Reply:
Is support for `TyAlias` for `structs` something that can be done in a follow-up PR?

Reply:
Issues created in https://github.com/immunant/c2rust/issues/926